docs/guide-disappearance.md

```diff
@@ -3,93 +3,135 @@ id: guide-disappearance
 title: Appearance and Disappearance
 ---
 
-Sometimes you need to test that an element is present and then disappears or
-vice versa.
+Sometimes you need to test that an element is present and then disappears or the
+other way around.
 
 ## Waiting for appearance
 
-If you need to wait for an element to appear, the [async wait
-utilities][async-api] allow you to wait for an assertion to be satisfied before
-proceeding. The wait utilities retry until the query passes or times out.
+If you need to wait for an element to appear, [`findBy`][find-by] and the [async
+wait utility][async-api] `waitFor` allow you to wait for an assertion to be
+satisfied before proceeding. The wait utilities retry until the query passes or
+times out.
 
-```jsx
+Wait for appearance and return the element using `findBy`:
+
+```javascript
+test('movie title appears', async () => {
+  // element is initially not present...
```

Comment:
Maybe add an extra assertion to check that it's not present initially?
```js
// element is initially not present...
expect(movieTitle).not.toBeInTheDocument()
```

Reply:
Maybe, but I feel that isn't entirely relevant and detracts a little from what we're trying to show in this example. Also, asserting that things don't exist is covered further down on the page.

If we did do that though, there should also be code here to do something that would make the title appear after asserting it isn't here, and I'm not sure it makes sense to get into that here.

We could add a comment saying as much after the assertion, like this:

```suggestion
  expect(screen.queryByText('the lion king')).not.toBeInTheDocument()
  
  // pretend we do something here that should make the title appear
```

But I think it might be better to alter the comment in the test to something that makes it clearer that this would be part of a longer test:

```suggestion
  // ...code setting up a page that will eventually contain a movie title

```

Do you think the last suggestion would help, or is there maybe something else we could do to alleviate the confusion or lack of clarity we are hoping to fix by adding this assertion?

Reply:
I think the first suggestion is much clearer instead of changing the comment tbh

Reply:
On the other hand, you're right with not wanting to have things in here that you'll explain later, so maybe the changed comment is indeed a better option to go with here...

Maybe we can make a full example (like the second suggestion) later on the page? 🤔

Reply:
>I think the first suggestion is much clearer instead of changing the comment tbh

Fair! In your mind, what is it that is unclear that adding this assertion solves? Or is there something else that makes it so you want it to be added?

Reply:
On the other hand, you're right with not wanting to have things in here that you'll explain later, so maybe the changed comment is indeed a better option to go with here...

Maybe we can make a full example (like the second suggestion) later on the page? 🤔

Reply:
I think there's value in keeping the first examples minimal, without having to decipher fake application code. Expanding on the comment and making clear this is a snippet of a larger test should be good.